x-pack/plugins/integration_assistant/server/graphs/ecs/pipeline.ts

```diff
@@ -0,0 +1,182 @@
+  try {
+    const keyList = key.split('.');
+    let value: any = samples;
+    for (const k of keyList) {
+      if (value === undefined || value === null) {
+        return null;
+      }
+      value = value[k];
+    }
+    return value;
+  } catch (e) {
+    throw e;
+  }
+}
+
+function getEcsType(ecsField: ECSField, ecsTypes: Record<string, string>): string {
+  const ecsTarget = ecsField.target;
+  return ecsTypes[ecsTarget];
+}
+
+function getConvertProcessorType(expectedEcsType: string): string {
+  if (expectedEcsType === 'long') {
+    return 'long';
+  }
+  if (['scaled_float', 'float'].includes(expectedEcsType)) {
+    return 'float';
+  }
+  if (expectedEcsType === 'ip') {
+    return 'ip';
+  }
+  if (expectedEcsType === 'boolean') {
+    return 'boolean';
+  }
+  return 'string';
+}
+
+function needsTypeConversion(sample: any, expected: string): boolean {
+  if (sample === null || sample === undefined) {
+    return false;
+  }
+
+  if (expected === 'ip') {
+    return true;
+  }
+
+  if (expected === 'boolean' && typeof sample !== 'boolean') {
+    return true;
+  }
+
+  if (['long', 'float', 'scaled_float'].includes(expected) && typeof sample !== 'number') {
+    return true;
+  }
+
+  if (
+    ['keyword', 'wildcard', 'match_only_text', 'constant_keyword'].includes(expected) &&
+    !(typeof sample === 'string' || Array.isArray(sample))
+  ) {
+    return true;
+  }
+
+  // If types are anything but the above, we return false. Example types:
+  // "nested", "flattened", "object", "geopoint", "date"
+  return false;
+}
+
+function generateProcessors(ecsMapping: object, samples: object, basePath: string = ''): object[] {
+  const ecsTypes = ECS_TYPES;
+  const valueFieldKeys = new Set(['target', 'confidence', 'date_formats', 'type']);
+  const results: object[] = [];
+
+  for (const [key, value] of Object.entries(ecsMapping)) {
+    const currentPath = basePath ? `${basePath}.${key}` : key;
+
+    if (value !== null && typeof value === 'object' && value?.target !== null) {
+      const valueKeys = new Set(Object.keys(value));
+      if ([...valueFieldKeys].every((k) => valueKeys.has(k))) {
+        const processor = generateProcessor(
+          currentPath,
+          value as ECSField,
+          getEcsType(value as ECSField, ecsTypes),
+          getSampleValue(currentPath, samples)
+        );
+        results.push(processor);
+      } else {
+        results.push(...generateProcessors(value, samples, currentPath));
+      }
+    }
+  }
+  return results;
+}
+
+export function createPipeline(state: EcsMappingState): IngestPipeline {
+  const samples = JSON.parse(state.formattedSamples);
+
+  const processors = generateProcessors(state.currentMapping, samples);
+  // Retrieve all source field names from convert processors to populate single remove processor:
+  const fieldsToRemove = processors.filter((p: any) => p.convert).map((p: any) => p.convert.field);
```

Comment:
```suggestion
  const fieldsToRemove = processors.filter((p: any) => p.convert).map((p: any) => p.convert.field).filter(x => x != null);
```

Not sure if the syntax is exactly right. But the idea is to fiter out `null` values from this `fieldsToRemove` list so that the process with `field: null` shall not be created.

Currently there are instances where the following processor is created

```
      {
        "remove": {
          "field": null,
          "ignore_missing": true,
          "tag": "remove_fields"
        }
      },
```

Reply:
Fixed in https://github.com/elastic/kibana/pull/184296/commits/d238614874eeb26d7178d15a090cbe951357fc3c